### ISSUE TITLE
Concurrency problem creating dotnet tool manifests

### ISSUE THREAD

**Issue by ghost** [dotnet/sdk#30254]
### Describe the bug
The command `dotnet new tool-manifest` produces a log file at C:\Users\UserName\AppData\Local\Temp, for example `Microsoft.NET.Workload_20230201_114619.log`. When more than one process execute it at the same time, an exception with the message `it is being used by another process` is thrown.

We use Azure DevOps Server 2022, with own managed agents. Every machine has more than one agent.
We install dotnet tools in the folder of each agent, to avoid concurrency problems. However, dotnet command produces them...

Inspecting the content of the log file shown in the exeption, I can read `[2023-01-28 08:18:22.227] [00004BC8] Executing: "C:\Program Files\dotnet\dotnet.exe" new tool-manifest --force, PID: 19400, PPID: 20616`. In another file like this one, i can read `[2023-01-29 23:51:59.897] [00004758] Executing: "C:\Program Files\dotnet\dotnet.exe" test D:\Agents\6\_work\1240\s\Solution.sln --logger trx --results-directory D:\Agents\6\_work\_temp -c Release --no-build -v quiet, PID: 18264, PPID: 8952`. So i think every dotnet command would produce an error like the one I am reporting, which is quite frightening.

I wonder if we can avoid producing the log file. I did not found anything about it.

### To Reproduce
`dotnet new tool-manifest & dotnet new tool-manifest & dotnet new tool-manifest & dotnet new tool-manifest &`

It is not always produced.

### Exceptions (if any)
```
Unhandled exception: System.IO.IOException: The process cannot access the file 'C:\Users\UserName\AppData\Local\Temp\Microsoft.NET.Workload_20230201_114619.log' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.CreateText(String path)
   at Microsoft.DotNet.Installer.Windows.TimestampedFileLogger..ctor(String path, Int32 flushThreshold, String[] logPipeNames)
   at Microsoft.DotNet.Workloads.Workload.Install.NetSdkMsiInstallerClient.Create(Boolean verifySignatures, SdkFeatureBand sdkFeatureBand, IWorkloadResolver workloadResolver, INuGetPackageDownloader nugetPackageDownloader, VerbosityOptions verbosity, PackageSourceLocation packageSourceLocation, IReporter reporter, String tempDirPath, RestoreActionConfig restoreActionConfig)
   at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallerFactory.GetWorkloadInstaller(IReporter reporter, SdkFeatureBand sdkFeatureBand, IWorkloadResolver workloadResolver, VerbosityOptions verbosity, String userProfileDir, Boolean verifySignatures, INuGetPackageDownloader nugetPackageDownloader, String dotnetDir, String tempDirPath, PackageSourceLocation packageSourceLocation, RestoreActionConfig restoreActionConfig, Boolean elevationRequired)
   at Microsoft.DotNet.Workloads.Workload.List.WorkloadInfoHelper..ctor(VerbosityOptions verbosity, String targetSdkVersion, Nullable`1 verifySignatures, IReporter reporter, IWorkloadInstallationRecordRepository workloadRecordRepo, String currentSdkVersion, String dotnetDir, String userProfileDir, IWorkloadResolver workloadResolver)
   at Microsoft.DotNet.Cli.NewCommandParser.CreateHost(Boolean disableSdkTemplates, Boolean disableProjectContext, FileInfo projectPath, FileInfo outputPath, LogLevel logLevel)
   at Microsoft.DotNet.Cli.NewCommandParser.<GetCommand>g__GetEngineHost|11_0(ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.BaseCommand.CreateEnvironmentSettings(GlobalArgs args, ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.BaseCommand`1.InvokeAsync(InvocationContext context)
   at Microsoft.TemplateEngine.Cli.Commands.BaseCommand`1.Invoke(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.DotNet.Cli.Parser.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__17_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass15_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass7_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

**Comment by IDoNotHaveName** [comment]
Issue #15399 is similar but not the same error is thrown. The cause may be different.

**Comment by edespong** [comment]
I am having the same issue (same call stack), when running `dotnet new install mypackage::1.2.3 --force` on a GitLab build agent.

We are not running any builds or other commands in parallel on the same machine, though.

**Comment by rajofro** [comment]
I have the same error and call stack as ghost when running `dotnet new tool-manifest` in a gitlab pipeline

**Comment by DoNotNeedMyName** [comment]
Having the possibility of disabling the creation of the log file would be nice. Is it possible?

The workaround I implemented was making retries if the exception is produced.

**Comment by baronfel** [comment]
I think per #15399 this is something that @SeanKilleen fixed for us that should release in the servicing updates to the SDK this month. Can you wait for that release and then reverify?

**Comment by baronfel** [comment]
you're right @KalleOlaviNiemitalo - I'm thinking of https://github.com/dotnet/sdk/pull/28736, a different issue that the templating team merged around lazily creating the workload resolver (a component that's used when doing project evaluations) that also released in 7.0.200. In any case, this should be resolved now.
